### DISCUSSION TITLE
ECONNRESET error without error.response.retryCount set

### DISCUSSION THREAD

**Discussion by rarkins** [sindresorhus/got#2166] [Q&A] [answered]
We are seeing intermittent `ECONNRESET` errors from registry.npmjs.org and I added additional debugging logs to check if they were being retried (I suspected not). It turns out that the `error.response` object does not have the `retryCount` field defined. The timings fields show 29s and `err.statusCode` is not set.

Considering that `ECONNRESET` _should_ be retried according to [the docs](https://github.com/sindresorhus/got/tree/v11#retry), what would you recommend as the next step? e.g. ways to check when an error's thrown that I haven't accidentally overridden the default `retry` settings?

**Comment by rarkins** [comment, marked as answer]
I guess there was no response, so therefore no error.response? But in this case is the retry information lost?

**Reply by szmarczak** [reply]
You can also access `error.request.retryCount`

**Reply by rarkins** [reply]
Thank you, I can confirm that I'm seeing confirmation of the retries in our logs now.
